src/run_sign.py

```diff
@@ -15,13 +15,19 @@
 from sign_workflow.signer import Signer
 from system import console
 
-ACCEPTED_SIGNATURE_FILE_TYPES = [".sig"]
+ACCEPTED_SIGNATURE_FILE_TYPES = [".sig", ".asc"]
+
+
+def null_str(val):
+    if not val:
+        return None
+    return val
 
 
 def main():
     parser = argparse.ArgumentParser(description="Sign artifacts")
     parser.add_argument("target", type=Path, help="Path to local manifest file or artifact directory.")
-    parser.add_argument("--component", nargs="?", help="Component name")
+    parser.add_argument("--component", nargs="?", type=null_str, help="Component name")
     parser.add_argument("--type", nargs="?", help="Artifact type")
     parser.add_argument("--sigtype", choices=ACCEPTED_SIGNATURE_FILE_TYPES, help="Type of Signature file", default=".asc")
```

Comment:
Could you also change the parser to pick default in case a null or a None is passed?

Reply:
The default would not be picked at all if calling from https://github.com/opensearch-project/opensearch-build/blob/main/vars/signArtifacts.groovy, because we specified `--component=${args.component}`. In that case, even we leave the `args.component` empty or null, parser will take it as a `string` type with empty content or string `null` . 
After my change, the argument type will be set as `NoneType` so still no default would be needed.

Reply:
Yes, you are correct. Maybe we should change to code such that if the value being passed is 'null' or 'None', the default value gets picked automatically. That way, anyone calling the script with null or None should be able to run with defaults. Thoughts?

Reply:
I think my way of resolving it is quite same as the circumstance you are saying. If any `null` or empty string passed in, parser will set its type to be `None`; not really necessary to have a default.

Reply:
Apologies for the confusion there. I meant for other parameters as well and default value in case of `--sigType`